node/clients/search-graphql.ts

```diff
@@ -1,18 +1,22 @@
-import { AppGraphQLClient, IOContext, InstanceOptions } from "@vtex/api";
+import { IOContext, InstanceOptions } from "@vtex/api";
+import { GraphQLServer } from './graphql-server';
+
 import { pathOr } from "ramda";
 
-export class SearchResolver extends AppGraphQLClient {
+const extensions = {
+  persistedQuery: {
+    provider: 'vtex.search-graphql@0.x',
+    sender: 'vtex.search@0.x',
+  },
+}
+
+export class SearchGraphql extends GraphQLServer {
```

Comment:
maybe by extending GraphQLServer, we could `super` the provider and sender, what do you think ?

Reply:
at least the sender is already defined, right ?

Reply:
I don't know if this is a good idea because in this same client we can do things other than query, for example, I can create a method for calling `/meta` not that makes sense do it on this scenario, but only to show that `sender and provider` is something that belongs to the query not to the entire graphql-server client.

Reply:
Not sure if that should be an inheritance or a composition. I'm thinking forward and I see our `{Service}GraphQLClient` with two possible ways of communication, direct to the service pod or via GraphQLServer.

Is that reasonable? If so, I don't like the idea of having two different clients to accomplish that. Those clients should have a `this.graphql` (as usual), a `this.graphqlServer` and someway to choose which one will be used.

For now I think the client implementer should decide how to deal with that choice. Maybe different methods, maybe arguments, etc. 

In the future, when we start generating a client based on the schema, we should provide a standard way to do it.

Reply:
@brunoabreu I don't think that using `graphql-server` as a way of communication with graphql apps outside of the domain of `Graphql Interface Apps` makes sense for now. But I agree that as this feature becomes more used for more things than search we have to think in a better interface than this one.

Reply:
Right. I just brought it to our attention because you said what we do here will be be an example for other projects.